src/speed-index.js

```diff
@@ -33,14 +33,44 @@ function calculateFrameProgress(current, initial, target) {
 	return progress;
 }
 
-function calculateVisualProgress(frames) {
+function calculateProgressBetweenFrames(frames, lowerBound, upperBound, threshold, getProgress, setProgress) {
+	if (threshold === 0) {
+		frames.forEach(frame => setProgress(frame, getProgress(frame)));
+		return;
+	}
+
+	const lowerFrame = frames[lowerBound];
+	const upperFrame = frames[upperBound];
+
+	const lowerProgress = getProgress(lowerFrame);
+	const upperProgress = getProgress(upperFrame);
+
+	setProgress(lowerFrame, lowerProgress);
+	setProgress(upperFrame, upperProgress);
+
+	if (Math.abs(lowerProgress - upperProgress) < threshold) {
```

Comment:
yeah I like it, but it'll likely erase a lot of the perf win if it's just straight up `&&`. In the CNN case, the same exact image is used across ~10s of the trace which accounts for ~30 frames. Thoughts on combining progress and time? 

wdyt of using that fun exponentiation window size function from TTFI?
```
allowable percent change = f(time elapsed in seconds) = 4 * e^(-.69 * t) + 1
f(0) = 4 + 1 = 5% allowed change when elapsed time is 0s
f(1) = 2 + 1 = 3% allowed change when elapsed time is 3s
f(∞) = 0 + 1 = 1% allowed change when elapsed time is very large
```